x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsResizeIntegTests.java

```diff
@@ -0,0 +1,141 @@
+import static org.elasticsearch.xpack.core.searchablesnapshots.MountSearchableSnapshotRequest.Storage;
+import static org.hamcrest.Matchers.equalTo;
+
+@ESIntegTestCase.ClusterScope(numDataNodes = 1)
+public class SearchableSnapshotsResizeIntegTests extends BaseFrozenSearchableSnapshotsIntegTestCase {
+
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        createRepository("repository", FsRepository.TYPE);
+        assertAcked(
+            prepareCreate(
+                "index",
+                Settings.builder()
+                    .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), 0)
+                    .put(INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), 2)
+                    .put(INDEX_NUMBER_OF_ROUTING_SHARDS_SETTING.getKey(), 4)
+                    .put(INDEX_SOFT_DELETES_SETTING.getKey(), true)
+            )
+        );
+        indexRandomDocs("index", scaledRandomIntBetween(0, 1_000));
+        createSnapshot("repository", "snapshot", List.of("index"));
+        assertAcked(client().admin().indices().prepareDelete("index"));
+        mountSnapshot("repository", "snapshot", "index", "mounted-index", Settings.EMPTY, randomFrom(Storage.values()));
+        ensureGreen("mounted-index");
+    }
+
+    @After
+    @Override
+    public void tearDown() throws Exception {
+        assertAcked(client().admin().indices().prepareDelete("mounted-*"));
+        assertAcked(client().admin().cluster().prepareDeleteSnapshot("repository", "snapshot").get());
+        assertAcked(client().admin().cluster().prepareDeleteRepository("repository"));
+        super.tearDown();
+    }
+
+    public void testShrinkSearchableSnapshotIndex() {
+        final IllegalArgumentException exception = expectThrows(
+            IllegalArgumentException.class,
+            () -> client().admin()
+                .indices()
+                .prepareResizeIndex("mounted-index", "shrunk-index")
+                .setResizeType(ResizeType.SHRINK)
+                .setSettings(
+                    Settings.builder()
+                        .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), 0)
+                        .put(INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), 1)
+                        .build()
+                )
+                .get()
+        );
+        assertThat(exception.getMessage(), equalTo("can't shrink searchable snapshot index [mounted-index]"));
+    }
+
+    public void testSplitSearchableSnapshotIndex() {
+        final IllegalArgumentException exception = expectThrows(
+            IllegalArgumentException.class,
+            () -> client().admin()
+                .indices()
+                .prepareResizeIndex("mounted-index", "split-index")
+                .setResizeType(ResizeType.SPLIT)
+                .setSettings(
+                    Settings.builder()
+                        .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), 0)
```

Comment:
NIT: I added `org.elasticsearch.snapshots.AbstractSnapshotIntegTestCase#indexSettingsNoReplicas` at one point for this because it's such a common kind of setting in tests :)